sphinx-search-as-you-type.rst

```diff
@@ -0,0 +1,147 @@
+All of my search related work can be seen in the `In-Doc Search UI Project Board`_.
+
+Background
+----------
+
+The search code was originally contributed by `Rob Hudson`_,
+`back in 2013`_ and then improved upon by other contributors.
+It was greatly improved and upgraded by `Safwan Rahman`_ during :doc:`GSoC'18 <search-improvements>`.
+Continuing on the same path,
+I have implemented some new features on top of the existing search backend.
+
+New Features
+------------
+
+During the GSoC period, I have worked on the following features:
+
+- **In-Doc Search UI**: `readthedocs-sphinx-search`_ is a sphinx extension that adds a clean and minimal
+  search UI to your docs. It supports "search as you type" feature.
+  So, it is now possible to get instant results without being redirected to any other page.
+  Read the docs `here`_.
+
+.. figure:: /_static/in-doc-search-demo.gif
+    :width: 100%
+    :align: center
+    :target: /_static/in-doc-search-demo.gif
+    :alt: In-doc search ui demo
+
+    In-Doc Search UI Demo
+
+- **Multiple Hits Per Search Result**: This is one of the highly requested features.
+  We now support search results from the sections of the docs, clicking on which will take you
+  to that particular section and not just to the top of the result page.
+
+- **Code Search**: We now support code search. If you want to search a particular function
+  or an API endpoint -- you can just type your query and you will find it in the results.
+  Eg: ``api/v3/`` or ``module.function``.
+
+- **Search Analytics**: We now have support for search analytics.
+  These analytics makes it easy to know what the users are looking for in your documentation.
+  You can see these analytics in your project admin dashboard.
+  Currently, this feature is in beta state and is available under a `feature flag`_.
+  We plan to make this available for everyone soon.
+  If you want to test this feature out and help giving us feedback,
+  please contact us via `GitHub issues`_.
+
+.. figure:: /_static/search-analytics-demo.png
+    :width: 60%
+    :align: center
+    :target: /_static/search-analytics-demo.png
+    :alt: Search analytics in project admin dashboard
+
+    Search analytics demo dashboard
+
+What Next?
+----------
+
+We don't intend to stop just yet.
+We are planning to work on some more cool features in the near future,
+some of which are:
+
+- **Search Facets**:
+  Facets can be used to make search more accurate.
+  For example: In `Celery docs`_, facets can be used to search inside `Kombu docs`_ for "serializers",
+  like ``subproject: kombu serializers``.
+  (`readthedocs/readthedocs.org#5966`_)
+- **Search Results Ordered By Most Viewed Pages**:
+  It would be much more useful if the most viewed pages are shown first in the search results.
+  (`readthedocs/readthedocs.org#5968`_)
+- **Search Inside Sections**:
+  It would be good if users have the option to get the
+  search results from a particular section of the documentation.
```

Comment:
From this description, this isn't totally clear what it would do.

Reply:
I added an example.